csrc/id_model/indexing.cpp

```diff
@@ -303,6 +312,112 @@ class AllocationDomainSetup : private kir::IrVisitor {
     return IndexingAllocationInfo{actual_allocation_domains, actual_strides};
   }
 
+  // Reorder non-logical allocation domains to follow the ordering of
+  // the logical domain. This is necessary when an allocation domain
+  // includes a vectorized loop iter domain since it must be at the
+  // innermost position but that may not be the case in the loop
+  // domain. Not strictly necessary otherwise, but this should also
+  // minimize the deviation from the old indexing scheme which always
+  // uses the logical domain to index.
+  //
+  // Returns reordered allocation domains if reordering is done.
+  std::optional<std::vector<IterDomain*>> reorderAllocationDomains(
+      const TensorView* tv,
+      const std::vector<IterDomain*>& allocation_domains) const {
+    // Don't change the set allocation domain. Ignore allocation
+    // domains of non-global tensors. For example,
+    // AliasTest.NotAllOutputAlias_Reduction has a tensor, tv6, that
+    // is a Local tensor with CA position of 4 but has an allocation
+    // domain that's just a permutation of its logical domain. Such
+    // invalid allocations need to be ignored.
+    if (tv->hasAllocation() && tv->getMemoryType() == MemoryType::Global) {
+      return std::nullopt;
+    }
```

Comment:
> Ignore allocation domains of non-global tensors.

Is this because global stores must always already have their vectorization axes innermost, or is there another reason?

Reply:
What we are doing here is inferring the right allocation domain of a given tensor. If a tensor already has a set allocation domain, I thought no inference should be done and just the set allocation domain should be used. So, originally, I didn't have `tv->getMemoryType() == MemoryType::Global`, but whenever a tensor has an allocation domain, no reordering should be done.

However, sometimes I found Local/Shared tensors also have allocation domains, and some of which don't seem to make sense, so this condition is just to ignore set allocation domains for non-global tensors.

This is an inference that I'm trying to match what the existing indexing implicitly assumes. It's not ideal because there's some ambiguity. There's also another "patch" that I'm going to create a PR for, but ideally, I think we should set the correct allocation domain for all tensors more explicitly. 

I consider this problem of figuring out the right allocation domain is a separate issue from indexing. I think we should refactor the allocation lowering pass such that each tensor gets the right allocation domain. These patches are not ideal, but for the time being, they should get us feature parity with the existing indexer.